html/captive-portal/lib/captiveportal/PacketFence/Controller/Authenticate.pm

```diff
@@ -224,9 +225,29 @@ sub postAuthentication : Private {
     $c->forward('setUnRegDate');
     $info->{source} = $source_id;
     $info->{portal} = $profile->getName;
+    $c->forward('checkIfTlsEnrollment');
     $c->forward('checkIfProvisionIsNeeded');
 }
 
+=head2 checkIfTlsEnrollment
+
+Checked to see if the provisioner should redirect you to TLS enrollment
+
+=cut
+
+sub checkIfTlsEnrollment : Private {
+    my ($self, $c) = @_;
+    my $portalSession = $c->portalSession;
+    my $mac = $portalSession->clientMac;
+    my $profile = $c->profile;
+    if (defined( my $provisioner = $profile->findProvisioner($mac))) {
+        if($provisioner->getPkiProvider) {
+            $c->session->{info} = $c->stash->{info};
+            $c->detach(tlsprofile => 'index');
```

Comment:
We usually have as a standard to call it with the right case.

In this case I think it should be TLSProfile